libraries/chain/apply_context.cpp

```diff
@@ -144,10 +144,12 @@ void apply_context::exec_one()
       r.auth_sequence[auth.actor] = next_auth_sequence( auth.actor );
    }
 
+   auto input_hash = digest_type::hash(act->data);
    if( control.is_builtin_activated( builtin_protocol_feature_t::action_return_value ) ) {
-      r.return_value.emplace( std::move( action_return_value ) );
+      auto output_hash = digest_type::hash(action_return_value);
+      r.act_digest = merkle({r.act_digest, output_hash});
```

Comment:
An `else` could be added so `r.act_digest` is not computed above if not needed.

Reply:
Oh, wait I read that as `{input_hash, output_hash}`. `input_hash` is currently not used.